integreat_chat/chatanswers/views.py

```diff
@@ -46,107 +46,120 @@ def search_documents(request):
+    ):
         data = json.loads(request.body)
         language_service = LanguageService()
-        if ("source_language" not in data or
-            "target_language" not in data or
-            "message" not in data
-            ):
-            result = {"status":"error"}
+        if (
+            "source_language" not in data
+            or "target_language" not in data
+            or "message" not in data
+        ):
+            result = {"status": "error"}
         else:
             result = {
                 "translation": language_service.translate_message(
-                    data["source_language"],
-                    data["target_language"],
-                    data["message"]
+                    data["source_language"], data["target_language"], data["message"]
                 ),
                 "target_language": data["target_language"],
-                "status": "success"
+                "status": "success",
             }
     return JsonResponse(result)
 
+
 @csrf_exempt
 def extract_answer(request):
     """
     Extract an answer for a user query from Integreat content. Expects a JSON body with message
     and language attributes
     """
     result = {}
-    if request.method in ('POST') and request.META.get('CONTENT_TYPE').lower() == 'application/json':
+    if (
+        request.method in ("POST")
+        and request.META.get("CONTENT_TYPE").lower() == "application/json"
+    ):
         data = json.loads(request.body)
-        if ("language" not in data or
-            "region" not in data or
-            "message" not in data
-            ):
-            result = {"status":"error"}
+        if "language" not in data or "region" not in data or "message" not in data:
+            result = {"status": "error"}
         else:
             language_service = LanguageService()
             if data["language"] not in settings.RAG_SUPPORTED_LANGUAGES:
                 rag_language = settings.RAG_FALLBACK_LANGUAGE
             else:
                 rag_language = data["language"]
-            if message_language := language_service.classify_language(data["language"], data["message"]) != data["language"]:
+            if (
+                message_language := language_service.classify_language(
+                    data["language"], data["message"]
+                )
+                != data["language"]
+            ):
                 message = language_service.translate_message(
-                    message_language,
-                    rag_language,
-                    data["message"]
+                    message_language, rag_language, data["message"]
                 )
             else:
                 message = data["message"]
 
             answer_service = AnswerService(data["region"], rag_language)
             if answer_service.needs_answer(data["message"]):
                 if settings.RAG_QUERY_OPTIMIZATION:
-                    message = answer_service.optimize_query_for_retrieval(message)
```

Comment:
If the `optimize_query_for_retrieval()` is not needed anymore, we can directly remove it from the AnswerService?

Reply:
Done. Can we merge this?